### ISSUE TITLE
CNTK and Tensorflow does not agree on pre-trained InceptionResNetV2

### ISSUE THREAD

**Issue by hvpham** [keras-team/keras-applications#12]
I have been trying to test the pre-trained weight for **InceptionResNetV2** on **CNTK** and **Tensorflow** backends on the ImageNet validation images with a single crop. And **Tensorflow** and **CNTK** do not agree with each other while **Tensorflow** and **Theano** do produce exactly the same output (It is mentioned on the application website that this model is compatible with all three backends). I attached the demo code that will recreate the problem on some selected images from the dataset (This is on Keras 2.1.6, Tensorflow 1.8.0, CNTK 2.5.1, Theano 1.0.1):

[inceptionResNetV2Demo.zip](https://github.com/keras-team/keras-applications/files/2094830/inceptionResNetV2Demo.zip)

I have also tested **DenseNet** (121, 169, 201), all variations output different prediction (on some images) between CNTK and Tensorflow/Theano. One interesting thing to note is that **VGG16** and **VGG19** are claimed to be compatible with Tensorflow and Theano only actually work with CNTK also (produced the same outputs as the other two).

There is another thing to note is that the overall accuracy of the model on the single cropped validation ImageNet 2012 dataset is only around **80-82%** top-5. The Tensorflow pre-trained weight report accuracy in the **95%** region.

Attached are the differences in predicted labels (Using pre-trained InceptionResNetV2 and between Tensorflow and CNTK):

![00015787-parachute](https://user-images.githubusercontent.com/9290537/41303389-1f071856-6e3b-11e8-91be-890eb6208889.png)
![00035325-microphone](https://user-images.githubusercontent.com/9290537/41303390-1f158a3a-6e3b-11e8-9be8-3ac04c4c913b.png)
![00042222-basketball](https://user-images.githubusercontent.com/9290537/41303391-1f23a5d4-6e3b-11e8-841f-d7ad2bd449b3.png)
![00047143-prayer_rug](https://user-images.githubusercontent.com/9290537/41303392-1f384052-6e3b-11e8-83b4-d8f1a06b08f9.png)
![04002704-petri_dish](https://user-images.githubusercontent.com/9290537/41303393-1f4c8f94-6e3b-11e8-8778-c8789beb9238.png)

**Comment by taehoonlee** [comment]
Thank you for the reports, @hvpham. I calculated squared errors between different backends' outputs with one single image:

|                   | TF & TH | TF & C |
|-------------------|---------|--------|
| VGG16             | 5e-16   | 3e-17  |
| DenseNet121       | 4e-16   | 5e-8   |
| MobileNet         | 1e-15   | 1e-8   |
| InceptionV3       | 1e-15   | 1e-7   |
| InceptionResNetV2 | 1e-15   | 1e-4   |

When a squared error of single softmax output is about 1e-8, a difference of performances on ImageNet validation set is usually less than 1%. But, InceptionResNetV2 with CNTK obviously produced different results as you told.

In results, they all came from the disagreement of BatchNormalization:
- TF/TH: `(x - mean) / C.sqrt(var + epsilon) * gamma + beta`,
- C: `(x - mean) / (C.sqrt(var) + epsilon) * gamma + beta`.

Due to the disagreement, the more BatchNormalization is done, the more the differences accumulate. But the disagreement has been resolved now (keras-team/keras#10427). **Thus, you can get the same outputs among all the backends if you use the bleeding edge of Keras.**
